ui/app/services/capabilities.ts

```diff
@@ -31,12 +40,28 @@ export default class CapabilitiesService extends Service {
     return assert('query object must contain "paths" or "path" key', false);
   }
 
-  /*
-  this method returns a capabilities model for each path in the array of paths
-  */
-  fetchMultiplePaths(paths: string[]): Promise<Array<CapabilitiesModel>> | AdapterError {
+  async fetchMultiplePaths(paths: string[]): MultipleCapabilities | AdapterError {
     try {
-      return this.request({ paths });
+      const resp: Array<CapabilitiesModel> = await this.request({ paths });
+      return resp.reduce((obj: MultipleCapabilities, model: CapabilitiesModel) => {
+        const path = paths.find((p) => model.path === p);
+        if (path) {
+          const { canCreate, canDelete, canList, canPatch, canRead, canSudo, canUpdate } = model;
+          obj[path] = { canCreate, canDelete, canList, canPatch, canRead, canSudo, canUpdate };
```

Comment:
Went the extra step of destructuring here so that just the `can` capabilities return. 
```js
// sample return
{
 "secret/data/my-secret": {
    canCreate: true,
    canDelete: false,
    canList: true,
    canPatch: true,
    canRead: true,
    canSudo: false,
    canUpdate: true,
 };
}
```

If we want to return the full capabilities model at some point (instead of an object with the capabilities) that's an easy refactor because the consumption of this method won't change. Retrieving the capability would still be `"secret/data/my-secret".canRead` etc.